src/emailjs-tcp-socket.js

```diff
@@ -386,6 +386,49 @@
     }
 
     function chromeShim() {
+
+        // setZeroTimeout slightly adapted from
+        // https://github.com/shahyar/setZeroTimeout-js (CC BY 3.0).
+        // Provides a function similar to setImmediate() on Chrome.
+        var setZeroTimeout = (function (w) {
+            var timeouts = [],
+            msg_name = 'asc0tmot',
+
+            // Like setTimeout, but only takes a function argument.  There's
+            // no time argument (always zero) and no arguments (you have to
+            // use a closure).
+            _postTimeout = function (fn) {
+                timeouts.push(fn);
+                postMessage(msg_name, '*');
+            },
+
+            _handleMessage = function (event) {
+                if (event.source === w && event.data === msg_name) {
+                    if (event.stopPropagation) {
+                        event.stopPropagation();
+                    }
+                    if (timeouts.length) {
+                        try {
+                            timeouts.shift()();
+                        } catch (e) {
+                            // Throw in an asynchronous closure to prevent setZeroTimeout from hanging due to error
+                            setTimeout((function (e) {
+                                return function () {
+                                    throw e.stack || e;
+                                };
+                            }(e)), 0);
+                        }
+                    }
+                    if (timeouts.length) { // more left?
+                        postMessage(msg_name, '*');
+                    }
+                }
+            };
+
+            addEventListener('message', _handleMessage, true);
+            return _postTimeout;
+        }(window));
+
         TCPSocket = function(config) {
```

Comment:
AFAIU browsers don't actually let you queue after 0ms, they enforce some minimum like 4-15ms. Having this kind of delay between each read call sounds like it could be bad news for throughput, especially when downloading larger attachments I guess.

I'll just mention some random benchmark I found: http://stackoverflow.com/questions/18826570/settimeout0-vs-window-postmessage-vs-messageport-postmessage

    2000 iterations of setTimeoutPM took 81 milliseconds.
    2000 iterations of setTimeout(0) took 10589 milliseconds.

`PM` is postMessage. This is a bit out of date, but all benchmarks I saw roughly said "setTimeout is super slow".

I haven't benchmarked anything in this particular case though, just wanted to be on the safe side.